_posts/2019-08-26-how-to-use-gruntwork-infrastructure-as-code-library.adoc

```diff
@@ -640,140 +640,124 @@ next two sections of the guide will walk you through how to use each of these.
 
-.infrastructure-modules/networking/vpc-app/main.tf
+.terraform-aws-service-catalog/modules/networking/vpc/main.tf
 [source,hcl]
 ----
-provider "aws" {
-  # The AWS region in which all resources will be created
-  region = var.aws_region
-
-  # Require a 2.x version of the AWS provider
-  version = "~> 2.6"
+terraform {
+  required_version: ">= 0.12.26"
 
-  # Only these AWS Account IDs may be operated on by this template
-  allowed_account_ids = [var.aws_account_id]
+  required_providers {
+    aws = {
+      source = "hashicorp/aws"
+      version = "~> 2.69"
+    }
+  }
 }
 ----
 
-This configures the AWS provider as follows:
-
-Use a specific AWS region::
-  The AWS region is configured via the `aws_region` input variable (you'll declare this shortly). This allows you to
-  deploy this module in multiple regions.
+This does the following:
 
 Pin the AWS provider version::
-  The code above ensures that you always get AWS provider version `2.x` and won't accidentally get version `3.x` in the
-  future, which would be backwards incompatible. We recommend pinning the versions for all providers you're using.
+  This pins the AWS provider version, ensuring that you always get AWS provider version `2.x` and won't accidentally get
+  version `3.x` in the future, which would be backward incompatible. We recommend pinning the versions for all
+  providers you're using.
 
-Pin AWS account IDs::
-  The code above will only allow you to run it against the AWS account with ID passed in via the `aws_account_id` input
-  variable (you'll declare this shortly). This is an extra safety measure to ensure you don't accidentally authenticate
-  to the wrong AWS account while deploying this code—e.g., so you don't accidentally deploy changes intended for
-  staging to production (for more info on working with multiple AWS accounts, see
-  link:/guides/foundations/how-to-configure-production-grade-aws-account-structure[How to Configure a Production Grade AWS Account Structure]).
+Pin the Terraform version::
+  The code above will ONLY allow you to run it with a specific Terraform version. This is a safety measure to ensure
+  you don't accidentally pick up a new version of Terraform until you're ready. This is important because (a) Terraform
+  is a pre 1.0.0 tool, so even patch version number bumps (e.g., `0.12.6` -> `0.12.7`) are sometimes backwards
+  incompatible or buggy and (b) once you've upgraded to a newer version, Terraform will no longer allow you to deploy
+  that code with any older version. For example, if a single person on your team upgrades to `0.12.7` and runs `apply`,
+  then you'll no longer be able to use the state file with `0.12.6`, and you'll be forced to upgrade everyone on your
+  team and all your CI servers to `0.12.7`. It's best to do this explicitly, rather than accidentally, so we recommend
+  pinning Terraform versions.
```

Comment:
This part is incorrect, as the service catalog module has a relaxed terraform version `>= 0.12.26` and not a strict `=` setup.

The advice is relevant though, and I would recommend having version locking in the root module calling the service catalog.